src/logic/calculate.js

```diff
@@ -0,0 +1,31 @@
+import operate from './operate';
+
+function calculate(calculatorData, buttonName) {
+  let { total, next, operation } = { ...calculatorData };
+
+  switch (buttonName) {
+    case '+/-':
+      if (next) {
+        next = (next * -1).toString();
+      } else if (total) {
+        total = (total * -1).toString();
+      }
+      break;
+    case 'AC':
+      total = '';
+      next = '';
+      operation = null;
+      break;
+    case '=':
+      total = operate(total, next, operation);
+      next = '';
+      operation = null;
+      break;
+    default:
+      total += buttonName;
+  }
+
+  return { total, next, operation };
+}
```

Comment:
Kindly ensure that all `buttonName` cases are covered. Since the `buttonName` represents all the available symbols in our calculator, you should ensure all operation cases (i.e. `+`, `-`, `/`, `x`) are covered in your switch case statement.